### DISCUSSION TITLE
Issues Adding MailJet for STMP

### DISCUSSION THREAD

**Discussion by heman22union** [dani-garcia/vaultwarden#4699] [Q&A] [answered]
I'm needing help setting up SMTP with vaultwarden. I'm trying to use mailjet. I have read the wiki but there isn't an example specific to mailjet. I'm using docker on casaos. I'm trying to configure these setting using the admin page on vaultwarden. I did some googling to get some examples on a mailjet config, but still can't get it to work.

Not sure if it's an issue with what I entered in the admin page or something else. The error message I get back make me think it's something wrong with now the admin page updates the JSON file? Here is a screenshot of the error and I didn't seen any issues on GitHub for this.

![image](https://github.com/dani-garcia/vaultwarden/assets/18408930/f75673e7-ab4a-4b7e-9cfe-37e0449e9b31)

**Comment by BlackDex** [comment]
A bit more information is needed, what have you tried, what did you filled in into the settings?
How does mailjet work with SMTP? What authentication is needed etc.. etc..

**Reply by heman22union** [reply]
Here is a screenshot of the settings. Should be what you are looking for?

![image](https://github.com/dani-garcia/vaultwarden/assets/18408930/999feca0-d7ec-4940-abb2-1c0e24d6291b)

**Reply by heman22union** [reply]
I also thing things using info from this reddit post - https://www.reddit.com/r/Bitwarden/comments/fbluky/struggling_with_email_on_bitwarden_self_hosted/

![image](https://github.com/dani-garcia/vaultwarden/assets/18408930/6747825a-ea1f-4deb-bebd-d6c555d6744a)

**Reply by heman22union** [reply]
Here is the domain ands senders page from MailJet. I have already setup my TXT records and everything needed on MailJet's side.

![image](https://github.com/dani-garcia/vaultwarden/assets/18408930/36846e9b-3816-47e3-8b27-2d20ca358f2c)

**Comment by BlackDex** [comment]
According to the mailjet smtp-relay page https://dev.mailjet.com/smtp-relay/overview/ you should use your api-key and api-secret-key so the first screenshot should be ok regarding that.

Further i see you use port 465 with starttls, that is probably not correct, either use port 587 or set the security to `force_tls` as describe also here: https://github.com/dani-garcia/vaultwarden/blob/a4c7fadbf4ffa242fb78ebd152080028e7f2993c/.env.template#L485-L492

I would suggest to switch to port 587 and try again. If that doesn't work, then also try to set the `Auth Mechanism` to `Plain` and see if that helps, but it shouldn't be needed.

**Reply by heman22union** [reply]
Thanks for the reply. I will try those. For the sake of the "not valid JSON" error, Where can I go to check the JSON file and provide the content of that file?

**Reply by BlackDex** [reply]
I'm not sure what json file you are revering too?

If you need more details during the debugging and in the logs i would suggest to enable `SMTP_DEBUG=true` and check the logs. Be careful with pasting those contents if you are planning to do so, it might contain your api-key/secret-key.

**Reply by heman22union** [reply, marked as answer]
Your recommendation for updating the Secure SMTP settings fixed it. I left it as 465 and updated it to force_tls.

Thanks for you help!
